Framework/Crystal/test/CalibrationHelpersTest.h

```diff
@@ -0,0 +1,287 @@
+#ifndef MANTID_CRYSTAL_CALIBRATIONHELPERSTEST_H_
+#define MANTID_CRYSTAL_CALIBRATIONHELPERSTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/DetectorInfo.h"
+#include "MantidCrystal/CalibrationHelpers.h"
+#include "MantidGeometry/IDTypes.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidGeometry/Instrument/ParameterMap.h"
+#include "MantidKernel/V3D.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using Mantid::Crystal::CalibrationHelpers;
+
+using namespace Mantid;
+using namespace Mantid::API;
+using namespace Mantid::Crystal;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+class CalibrationHelpersTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static CalibrationHelpersTest *createSuite() {
+    return new CalibrationHelpersTest();
+  }
+  static void destroySuite(CalibrationHelpersTest *suite) { delete suite; }
+
+  void test_fixUpSampleAndSourcePositions_moves_the_sample_and_the_source() {
+    // Create two identical workspaces
+    const auto wsOld =
+        WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(2, 1000,
+                                                                     true);
+    auto wsNew = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+        2, 1000, true);
+
+    const auto &positionSampleOld = wsOld->detectorInfo().samplePosition();
+    const auto &positionSourceOld = wsOld->detectorInfo().sourcePosition();
+
+    const auto &instNew = wsNew->getInstrument();
+    const double l1 = 10.0;
```

Comment:
Could you write this such that it is obvious that the new `l1` is different from the old one? For example, use `wsOld->detectorInfo().l1() - 1.23`;